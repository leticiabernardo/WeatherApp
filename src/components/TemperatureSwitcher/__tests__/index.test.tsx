import { render, cleanup, fireEvent } from '@testing-library/react';
import TemperatureSwitcher from '../index';

const props = {
  setMeasurementUnit: jest.fn(),
};

describe('temperature switcher render', () => {
  afterEach(cleanup);

  test('render the component', () => {
    const { container } = render(<TemperatureSwitcher {...props} />);
    expect(container).toMatchSnapshot();
  });

  test('should have the button to celsius', () => {
    const { getByText } = render(<TemperatureSwitcher {...props} />);

    expect(getByText('째C')).toBeInTheDocument();
  });

  test('should have the button to switch fahrenheit', () => {
    const { getByText } = render(<TemperatureSwitcher {...props} />);

    expect(getByText('째F')).toBeInTheDocument();
  });
});

describe('temperature switcher events', () => {
  afterEach(cleanup);

  test('should change to celsius', async () => {
    const { container, getByText } = render(
      <TemperatureSwitcher {...props} measurementUnit="fahrenheit" />
    );
    const button = getByText('째C');
    await fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });

  test('should change to fahrenheit', async () => {
    const { container, getByText } = render(<TemperatureSwitcher {...props} />);
    const button = getByText('째F');
    await fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});
