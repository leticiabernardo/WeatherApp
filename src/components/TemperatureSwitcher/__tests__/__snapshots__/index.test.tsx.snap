// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`temperature switcher events should change to celsius 1`] = `
<div>
  <span
    class="css-10kghbw"
  >
    <button
      class="chakra-button css-hu7fh3"
      type="button"
    >
      <span
        class="css-0"
        tabindex="0"
      >
        °C
      </span>
    </button>
    |
    <button
      class="chakra-button css-yqy1ty"
      type="button"
    >
      <span
        class="css-0"
        tabindex="0"
      >
        °F
      </span>
    </button>
  </span>
</div>
`;

exports[`temperature switcher events should change to fahrenheit 1`] = `
<div>
  <span
    class="css-10kghbw"
  >
    <button
      class="chakra-button css-yqy1ty"
      type="button"
    >
      <span
        class="css-0"
        tabindex="0"
      >
        °C
      </span>
    </button>
    |
    <button
      class="chakra-button css-hu7fh3"
      type="button"
    >
      <span
        class="css-0"
        tabindex="0"
      >
        °F
      </span>
    </button>
  </span>
</div>
`;

exports[`temperature switcher render render the component 1`] = `
<div>
  <span
    class="css-10kghbw"
  >
    <button
      class="chakra-button css-yqy1ty"
      type="button"
    >
      <span
        class="css-0"
        tabindex="0"
      >
        °C
      </span>
    </button>
    |
    <button
      class="chakra-button css-hu7fh3"
      type="button"
    >
      <span
        class="css-0"
        tabindex="0"
      >
        °F
      </span>
    </button>
  </span>
</div>
`;
